staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go

```diff
@@ -237,7 +243,16 @@ func (o *CreateDeploymentOptions) createDeployment() *appsv1.Deployment {
 	if o.Port >= 0 && len(deploy.Spec.Template.Spec.Containers) > 0 {
 		deploy.Spec.Template.Spec.Containers[0].Ports = []corev1.ContainerPort{{ContainerPort: o.Port}}
 	}
-	return deploy
+
+	cliLabels, err := cmdutil.ParseLabels(o.Labels)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range cliLabels {
+		deploy.Labels[k] = v
+	}
```

Comment:
OK, here we have a "problem": what happens if I specify something like "--label app=blo" while my app is called "bla"? This can generate inconsistencies, we should probably verify if k!="app" and then merge it into the map